extensions/sql-migration/src/wizard/accountsSelectionPage.ts

```diff
@@ -0,0 +1,112 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as azdata from 'azdata';
+import * as vscode from 'vscode';
+import { MigrationWizardPage } from '../models/migrationWizardPage';
+import { MigrationStateModel, StateChangeEvent } from '../models/stateMachine';
+import * as constants from '../models/strings';
+
+export class AccountsSelectionPage extends MigrationWizardPage {
+	private _azureAccountsDropdown!: azdata.DropDownComponent;
+	private _accountsMap!: Map<string, azdata.Account>;
+
+	// For future reference: DO NOT EXPOSE WIZARD DIRECTLY THROUGH HERE.
+	constructor(wizard: azdata.window.Wizard, migrationStateModel: MigrationStateModel) {
+		super(wizard, azdata.window.createWizardPage(constants.ACCOUNTS_SELECTION_PAGE_TITLE), migrationStateModel);
+		this._accountsMap = new Map();
+	}
+
+	protected async registerContent(view: azdata.ModelView): Promise<void> {
+		const form = view.modelBuilder.formContainer().withFormItems(
+			[
+				await this.createAzureAccountsDropdown(view)
+			]
+		);
+		await view.initializeModel(form.component());
+		await this.populateAzureAccountsDropdown();
+		return;
+	}
+
+	private async createAzureAccountsDropdown(view: azdata.ModelView): Promise<azdata.FormComponent> {
+
+		this._azureAccountsDropdown = view.modelBuilder.dropDown().withProperties({}).component();
+
+		this._azureAccountsDropdown.onValueChanged(async (value) => {
+			this.migrationStateModel.azureAccount = this._accountsMap.get(value.selected)!;
+		});
+
+		const addAccountButton = view.modelBuilder.button().withProperties<azdata.ButtonProperties>({
+			label: constants.ACCOUNT_ADD_BUTTON_LABEL,
+			width: '100px'
+		}).component();
+
+		addAccountButton.onDidClick(async (event) => {
+			await vscode.commands.executeCommand('workbench.actions.modal.linkedAccount');
+			await this.populateAzureAccountsDropdown();
+		});
+
+		const flexContainer = view.modelBuilder.flexContainer().withLayout({
```

Comment:
suggest putting .withLayout, .withItems and .component() calls each started on an indented newline.